### PR TITLE
Fix Issue https://github.com/piwik/piwik/issues/11410

### LINKED ISSUE #11410: Optional secure tracker cookie
I think this topic was discussed years ago, but I do get negativ security points via Mozillas observatory when delivering first party tracking cookies without the secure flag.

I think it should be possible to enable that plag on a per website basis. Most websites I setup are SSL only, a request to non encrypted pages gets redirected to ssl and that ist the recommended canonical url for every page. So I don't need any sharing of session tracking cookies between http and https.

### PR DESCRIPTION
**Pull request by ghost**
fixes #11410

### CODE REVIEW

File: js/piwik.js [hunk 1]
```diff
@@ -3223,7 +3225,14 @@ if (typeof window.Piwik !== 'object') {
...
                     (domain ? ';domain=' + domain : '') +
-                    (secure ? ';secure' : '');
+                    (isSecureCookie ? ';secure' : '');
+            }
+
+            /*
+             * Set cookie secure flag
+             */
+            function setSecureCookie(value) {
+                isSecureCookie = value == 1 ? true : false;
```
Comment (sgiehl) [inline js/piwik.js]: JSLint test fails here. Is there any reason for comparing `value` with `1`? If you want to ensure `isSecureCookie` contains a Boolean afterwards, you could also cast `value` to Boolean or simply use `isSecureCookie = !!value`, which should pass JSLint.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Besides the small change required to pass JSLint, the minified piwik.js needs to be built. But if you can't to that, we can do it before/after merging.

**Comment by sgiehl** [comment]
@dudu84 do you have time to make the mentioned changes?

**Comment by dudu84** [comment]
@sgiehl Yes, will to until tomorrow. Sorry for the delay.

File: js/piwik.js (hunk 1 above)
Comment (dudu84) [inline js/piwik.js]: @sgiehl Hi there! I did the changes you requested, including the minifying. I can see there on my piwik branch but they are not being reflected in the PR. Could you tell me why?

Comment (sgiehl) [inline js/piwik.js]: Can see them on the PR as well, but could you try to rebase your branch and rebuilt the minified file? It's currently conflicting with 3.x-dev branch.

**Comment by dudu84** [comment]
@sgiehl Hi Stefan, I did the rebase. But it is conflicting now. Could you shed a light, please?

**Comment by sgiehl** [comment]
@dudu84 seems your fork is not even with piwik repo. So you need to update your fork before doing the rebase. 
Alternatively revert your changes to the minified files and I will update them after merging...

**Comment by dudu84** [comment]
@sgiehl Thanks for the patience so far. Well I forked it again and applied the changes on a the update repo. Didn't know an easier way to do this, sorry. So I create a new PR: https://github.com/piwik/piwik/pull/11834

**Comment by sgiehl** [comment]
Ok. Thanks. I'll close this one.

**Pull request closed without merging**